### PR TITLE
Automating release distro build

### PR DESCRIPTION
**Pull request by dejanb**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -147,16 +147,16 @@
     <module>bom</module>
     <module>broker</module>
     <module>core</module>
+    <module>service-base</module>
```
Comment (sophokles73) [inline pom.xml]: is there a reason you need to re-order the modules?

File: example/src/main/assembly/hono.xml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            <directory>../adapters/mqtt-vertx/target/classes/META-INF/fabric8</directory>
+            <fileMode>0644</fileMode>
+            <directoryMode>0755</directoryMode>
+        </fileSet>
+
+    </fileSets>
+
+
+
+</assembly>
```
Comment (sophokles73) [inline example/src/main/assembly/hono.xml]: EOL missing

File: example/pom.xml [hunk 3]
```diff
@@ -74,7 +74,17 @@ Also contains a sample Docker Compose file (see &quot;docker&quot; folder) for s
       <resource>
         <directory>${project.basedir}/docker</directory>
         <filtering>true</filtering>
-        <targetPath>${project.build.directory}/hono</targetPath>
+        <targetPath>${project.build.directory}/hono/example/docker</targetPath>
```
Comment (sophokles73) [inline example/pom.xml]: If we change the path here we also need to adapt the *Getting Started* and deployment guides accordingly. I have no problem with that, but now that we are doing this, maybe we can use
`${project.build.directory}/deploy/docker`
`${project.build.directory}/deploy/kubernetes`
`${project.build.directory}/deploy/openshift`?

File: example/pom.xml [hunk 4]
```diff
@@ -74,7 +74,17 @@ Also contains a sample Docker Compose file (see &quot;docker&quot; folder) for s
...
+      </resource>
+      <resource>
+        <directory>${project.basedir}/openshift</directory>
```
Comment (sophokles73) [inline example/pom.xml]: move all deploy *src* scripts under a common base folder so that we need to define one resource here, e.g
keeping sources in
src/main/deploy/docker
src/main/deploy/kubernetes
src/main/deploy/openshift
and then just define
```
<resource>
  <directory>${project.basedir}/src/main/deploy</directory>
  <filtering>true</true>
  <targetPath>${project.build.directory}/deploy</targetPath>
</resource>
```
?

Comment (ppatierno) [inline example/pom.xml]: I'm ok with these changes but I'm starting to think that this project isn't an example anymore ... even if it's really is because it's one of the possible Hono deployments. Moving from using "example" to "deploy" inside a path as suggested by @sophokles73 let me think that. Maybe we should rename the entire project ?

File: example/pom.xml [hunk 5]
```diff
@@ -132,6 +143,27 @@ Also contains a sample Docker Compose file (see &quot;docker&quot; folder) for s
...
+                <phase>package</phase>
+                <goals>
+                    <goal>single</goal>
+                </goals>
+                <configuration>
+                  <descriptors>
+                    <descriptor>src/main/assembly/hono.xml</descriptor>
+                  </descriptors>
+                  <tarLongFileMode>posix</tarLongFileMode>
+                  <finalName>eclipse-hono-${project.version}</finalName>
```
Comment (ppatierno) [inline example/pom.xml]: if we stay that it's still an "example", maybe it should be reflected in the final tar name ? i.e. eclipse-hono-example-${project.version} ?

File: pom.xml (hunk 1 above)
Comment (dejanb) [inline pom.xml]: Yes, because we need artefacts(yml files) from those modules

File: example/pom.xml (hunk 3 above)
Comment (dejanb) [inline example/pom.xml]: I was thing to leave it as it is, but this makes more sense to me, so I'm for it

File: example/pom.xml (hunk 4 above)
Comment (dejanb) [inline example/pom.xml]: @sophokles73 Sounds good to me
@ppatierno I was thinking about creating new assembly module, but in a way ti is an example deployment so it can stay like this for now. We can change the name of the file.

Comment (ppatierno) [inline example/pom.xml]: I'm ok with that ... it was just a "philosophical" thought :-)

File: example/pom.xml (hunk 5 above)
Comment (sophokles73) [inline example/pom.xml]: Well, it's just the example deployment script. Still, it's a deployment script ... FMPOV we do not need to change the project's name. But I like the idea of putting `example` into the artifact name ...

File: pom.xml (hunk 1 above)
Comment (sophokles73) [inline pom.xml]: But why do you need to change the order of the modules in the POM file for that?

Comment (dejanb) [inline pom.xml]: These modules need to be built before 'example', that's all

Comment (sophokles73) [inline pom.xml]: changing the order in the pom does not have any influence on the order in which the projects are built AFAIK. The build order is determined by means of resolving dependencies between the modules at the beginning of the (reactor) build. These modules will be built before `example` simply because `example` depends on them (either directly or transitively) ...

Comment (sophokles73) [inline pom.xml]: can you please undo this change?

Comment (dejanb) [inline pom.xml]: You're right, I wasn't thinking properly. I need to add proper dependencies to the example module. I assume you want to keep modules in alphabetical order?

Comment (sophokles73) [inline pom.xml]: As expected, you immediately recognized the pattern ;-)

File: example/src/main/assembly/hono.xml (hunk 2 above)
Comment (dejanb) [inline example/src/main/assembly/hono.xml]: This is fixed now

File: pom.xml (hunk 1 above)
Comment (dejanb) [inline pom.xml]: All in order now :)

File: example/pom.xml [hunk 6]
```diff
@@ -67,6 +67,48 @@ Also contains a sample Docker Compose file (see &quot;docker&quot; folder) for s
       <groupId>org.eclipse.hono</groupId>
       <artifactId>hono-demo-certs</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.hono</groupId>
+      <artifactId>broker</artifactId>
```
Comment (sophokles73) [inline example/pom.xml]: there is no such artifact. It is called `hono-artemis` now ... this currently beaks the build on Travis.

**Comment by sophokles73** [comment]
LGTM, can you please squash and rebase before merging?
